python/pyspark/streaming/dstream.py

```diff
@@ -454,7 +454,10 @@ def reduceByWindow(self, reduceFunc, invReduceFunc, windowDuration, slideDuratio
         This is more efficient than `invReduceFunc` is None.
 
         @param reduceFunc:     associative reduce function
-        @param invReduceFunc:  inverse reduce function of `reduceFunc`
+        @param invReduceFunc:  inverse reduce function of `reduceFunc`; such that for all x,
+                               and invertible y:
+                               `invReduceFunc(reduceFunc(x, y), y) = x`
+                               Both `reduceFunc` and `invReduceFunc` should be associative.
```

Comment:
Are you sure about this? The inverse reduce of X+Y is Z-X which is not associative.
(a - b) - c =/= a - (b - c)